pkg/apis/pipeline/v1beta1/task_validation.go

```diff
@@ -498,31 +501,57 @@ func validateStepArrayUsage(step Step, prefix string, vars sets.String) *apis.Fi
 }
 
 func validateVariables(ctx context.Context, steps []Step, prefix string, vars sets.String) (errs *apis.FieldError) {
-	// validate that the variable name format follows the rules
-	// - Must only contain alphanumeric characters, hyphens (-), underscores (_), and dots (.)
-	// - Must begin with a letter or an underscore (_)
-	re := regexp.MustCompile(variableNameFormat)
+	// We've checked param name format. Now, we want to check if param names are referenced correctly in each step
+	for idx, step := range steps {
+		errs = errs.Also(validateStepVariables(ctx, step, prefix, vars).ViaFieldIndex("steps", idx))
+	}
+	return errs
+}
+
+// validateNameFormat validates that the variable name format follows the rules
```

Comment:
I feel you can move the comments to the regex itself?

Reply:
done